grails-testing-support/src/main/groovy/org/grails/testing/GrailsUnitTest.groovy

```diff
@@ -155,10 +156,20 @@ trait GrailsUnitTest {
             DeferredBindingActions.clear()
 
             this._grailsApplication = null
-
-            Promises.promiseFactory = null
-
+            cleanupPromiseFactory()
             Holders.clear()
         }
     }
+
+    private void cleanupPromiseFactory() {
+        ClassLoader classLoader = getClass().classLoader
+        if (ClassUtils.isPresent("grails.async.Promises", classLoader)) {
+            final String className = "grails.async.Promises"
+            final Field field = ClassUtils.forName(className, classLoader)
```

Comment:
You don't have to do this. You can just call `Promises.proiseFactory = null` inside the if condition

Reply:
Please note that the error does not happen when we call this method. I think we will see the same problem if we call `Promises.proiseFactory = null` because it won't be able to find the class `Promises` while compiling.